core/src/main/java/org/apache/druid/data/input/impl/JSONParseSpec.java

```diff
@@ -98,7 +106,7 @@ public boolean equals(final Object o)
       return false;
     }
     final JSONParseSpec that = (JSONParseSpec) o;
-    return Objects.equals(featureSpec, that.featureSpec);
+    return Objects.equals(featureSpec, that.featureSpec) && Objects.equals(keepNullColumns, that.keepNullColumns);
```

Comment:
Missing corresponding update to hashCode. Can you add an `EqualsVerifier` test for this class please.